src/ansible/roles/ad/tasks/main.yml

```diff
@@ -14,60 +15,42 @@
     type: string
     state: present
 
-- name: Install Active Directory Services
-  win_feature:
-    name: '{{ item }}'
-    include_management_tools: yes
-    include_sub_features: yes
-    state: present
-  with_items:
-  - AD-Domain-Services
-  - DNS
-
-- name: 'Create new AD forest {{ service.ad.domain }}'
-  win_shell: |
-    Import-Module ADDSDeployment
+- name: Detect cygwin
+  win_stat:
+    path: 'C:\cygwin64\etc'
+  register: cygwin
 
-    Install-ADDSForest                                                        \
-      -DomainName "{{ service.ad.domain }}"                                   \
-      -CreateDnsDelegation:$false                                             \
-      -DomainNetbiosName "{{ service.ad.netbios }}"                           \
-      -ForestMode "WinThreshold"                                              \
-      -DomainMode "WinThreshold"                                              \
-      -Force:$true                                                            \
-      -InstallDns:$true                                                       \
-      -NoRebootOnCompletion:$true                                             \
-      -SafeModeAdministratorPassword                                          \
-        (ConvertTo-SecureString '{{ service.ad.safe_password }}' -AsPlainText -Force)
-  register: installation
-  args:
-    creates: 'C:\Windows\NTDS'
+- name: Configure shell for cygwin
+  win_lineinfile:
+    path: 'C:\cygwin64\etc\nsswitch.conf'
+    line: "db_shell: /cygdrive/c/Windows/System32/WindowsPowerShell/v1.0/powershell.exe"
+  when: cygwin.stat.exists
+  register: configured_cygwin
 
-- name: Reboot machine
+- name: Reboot machine to apply changes in cygwin config
   win_reboot:
-  when: installation.changed
+  when: configured_cygwin.changed and skip_addc_install
+
+- name: 'Install AD server'
+  include_tasks: 'install.yml'
+  when: not skip_addc_install
+
+- name: Install management tools
+  win_feature:
+    name:
+    - RSAT-AD-Tools
+    include_sub_features: yes
+    include_management_tools: yes
 
 - name: Make sure Active Directory Web Services is running
   win_service:
     name: adws
     start_mode: auto
     state: started
 
-- name: Copy sudo schema to guest
-  win_copy:
-    src: '{{ item }}.schema'
-    dest: 'C:\{{ item }}.schema'
-  with_items:
-  - sudo
-
-- name: Install additional schemas
-  win_shell: |
-    ldifde -i -f C:\{{ item }}.schema -c dc=X {{ service.ad.suffix }} -b "Administrator" "{{ service.ad.netbios }}" "vagrant"
-  register: schema
-  failed_when: schema.rc != 0 and schema.stdout is not search('ENTRY_EXISTS')
-  changed_when: schema.rc == 0
-  with_items:
-  - sudo
+- name: 'Add sudo schema and possibly other'
+  include_tasks: 'schema.yml'
+  when: not skip_schema
 
 - name: Set Password Never Expires for system users
```

Comment:
This step has never ones failed before and now it takes 5 retries for each account. Please, debug it. It takes really long and it can not be because AD is not responsive, because each account takes 5 retries so it has to retry even when we were already able to modify the first user. Maybe try to reboot just before "Make sure Active Directory Web Services is running" step?

Reply:
I have created an issue for this https://github.com/SSSD/sssd-ci-containers/issues/52.
I will look in into it as a part of AD improvements. Please do not block this MR as others are waiting for it.